src/module.c

```diff
@@ -5617,6 +5623,11 @@ RedisModuleString *RM_CreateStringFromCallReply(RedisModuleCallReply *reply) {
     }
 }
 
+/* Modifies the user that RM_Call will use (e.g. for ACL checks) */
+void RM_SetContextModuleUser(RedisModuleCtx *ctx, const RedisModuleUser *user) {
```

Comment:
I still have trouble with this API name.
both the words "context" and "module" seem a bit excessive.

looking at another API that does a similar thing: `RM_AuthenticateClientWithUser`

so the difference is that one works on the client and the other on the context, and that one is more predominant than the other.
I think we can drop the word "Module", i think it is implied.
if we'll ever have a similar API that takes a non-module user, we'll call it SetContextUserName.
same as we have RM_GetCurrentUserName vs RM_GetModuleUserFromUser**Name**, and the similar RM_AuthenticateClientWith**ACL**User vs RM_AuthenticateClientWithUser

Reply:
Don't think we have a strong objection for renaming it to `RM_SetContextUser()` - will do it. 🙏